library/Vanilla/Utility/SchemaFilters.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license https://opensource.org/licenses/GPL-2.0 GPL-2.0
+ */
+
+namespace Vanilla\Utility;
+
+use Garden\Schema\ValidationField;
+
+/**
+ * A collection of field filter functions, compatible with Garden Schema.
+ */
+class SchemaFilters {
+
+    /**
+     * Attempt to decode a encoded string value into a more complex type.
+     *
+     * @param mixed $value Raw database value.
+     * @param ValidationField $field An object representing the Garden Schema field.
+     * @return mixed Unpacked attributes field value.
+     */
+    public static function encodeValue($value, ValidationField $field) {
+        if ($value === null || $value === '') {
+            $value = null;
+        } else {
+            $value = StringUtils::jsonEncodeChecked($value, JSON_UNESCAPED_SLASHES);
+        }
+        return $value;
+    }
+
+    /**
+     * Attempt to decode a encoded string value into a more complex type.
+     *
+     * @param mixed $value Raw database value.
+     * @param ValidationField $field An object representing the Garden Schema field.
+     * @return mixed Unpacked attributes field value.
+     */
+    public static function decodeValue($value, ValidationField $field) {
+        if ($value === null || $value === '') {
+            $value = null;
+        } elseif (is_string($value)) {
+            $value = json_decode($value, true);
+            if ($value === null) {
+                try {
+                    $value = unserialize($value);
```

Comment:
I'm curious why we're allowing for serialization in our pristine new model. Without this, doesn't this basically work out to an elaborate alias for json_decode? Are we planning to point this model at old, busted data? Again, not super bothered, but curious what the plans are here.

Reply:
> Are we planning to point this model at old, busted data?

Yep! Kinda, anyway. The aim here is to allow some kinda upgrade path for old models. For those, we'll allow reading of the existing serialized data, but force writing of JSON-encoded data.